File: src/main/resources/assets/minecraft/scripts/Render_TDT01-F3.js
var renderClass = "jp.ngt.rtm.render.VehiclePartsRenderer";

importPackage(Packages.com.github.ikoko125.banetsu2419_ns3000.model);
importPackage(Packages.com.github.ikoko125.banetsu2419_ns3000.render);

function init(modelSet, modelObject) {
    ModelTDT01F3.registerObjects(renderer);
}

function render(entity, pass, par3) {
    RenderTDT01F3.render(renderer, entity, pass)
}

File: src/main/kotlin/com/github/ikoko125/banetsu2419_ns3000/render/RenderTDT01F3.kt
package com.github.ikoko125.banetsu2419_ns3000.render

import com.github.ikoko125.banetsu2419_ns3000.model.ModelTDT01F3
import jp.ngt.rtm.entity.train.EntityBogie
import jp.ngt.rtm.render.VehiclePartsRenderer
import org.lwjgl.opengl.GL11

@Suppress("unused")
object RenderTDT01F3 {
    @JvmStatic
    fun render(renderer: VehiclePartsRenderer, entity: EntityBogie?, pass: Int) {
        if (pass != 0) return

        val model = ModelTDT01F3.getModel(renderer.modelName)!!
        val wheelRotation = renderer.getWheelRotationR(entity)

        GL11.glPushMatrix()
        model.body.render(renderer)

        GL11.glPushMatrix()
        renderer.rotate(wheelRotation, 'X', 0f, -0.54f, 1.05f)
        model.wheel1.render(renderer)
        GL11.glPopMatrix()

        GL11.glPushMatrix()
        renderer.rotate(wheelRotation, 'X', 0f, -0.54f, -1.05f)
        model.wheel2.render(renderer)
        GL11.glPopMatrix()

        GL11.glPopMatrix()
    }
}
